### PR TITLE
Support quantize with no eval function for easy model conversion

### PR DESCRIPTION
**Pull request by raghuramank100**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #24924 Serialization for nn.quantized.functional modules
* #24923 Support observer without any data calibration
* **#24922 Support quantize with no eval function for easy model conversion**
* #24804 Update mapping dictionary to support functionalmodules and pooling operations
* #24803 Handle empty qconfig for functional Modules
* #24789 Work around for bias quantization for conv and linear operators

Support quantize() with no eval function arguments so that one can create a quantized model without calibration

Differential Revision: [D16923670](https://our.internmc.facebook.com/intern/diff/D16923670/)

### CODE REVIEW

File: torch/quantization/quantize.py [hunk 1]
```diff
@@ -246,7 +246,8 @@ def quantize(model, run_fn, run_args, mapping=DEFAULT_MODULE_MAPPING):
     """
     model.eval()
     model = prepare(model)
-    run_fn(model, run_args)
+    if run_fn:
+        run_fn(model, run_args)
```
Comment (jerryzh168) [inline torch/quantization/quantize.py]: if you don't run it how do we get qparams?

Comment (raghuramank100) [reply torch/quantization/quantize.py]: I have the changes to observer to do that in a seperate diff: https://github.com/pytorch/pytorch/pull/24923.  This is being done to speed up creation of quantized model. Currently if I want to load a quantized model, I have to first call prepare and then convert. Eliminating the eval step speeds the process.

Comment (jerryzh168) [reply torch/quantization/quantize.py]: I see

**Comment by jerryzh168** [comment]
we can close this I think..

**Pull request closed without merging**